pennylane/math/multi_dispatch.py

```diff
@@ -154,7 +155,7 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-@multi_dispatch(argnum=[0])
+@multi_dispatch(argnum=[0, 1])

```

Comment:
Was this a bug? How exactly is this change affecting the behaviour of `qml.math.kron`?

Reply:
I guess kind of a bug?

In line 483 of `two_qubit_unitary.py`, we take `math.kron(math.eye(2), RYa)`, where, when jitting, `RYa` is abstract. But `math.eye(2)` is not abstract, and with `argnum=[0]`, `multi_dispatch` only looked at the first argument to set the interface, so it didn't use `jax.numpy`. This caused the error `TracerArrayConversionError: The numpy.ndarray conversion method __array__() was called on the JAX Tracer object...`